modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/Type.java

```diff
@@ -169,4 +169,16 @@ default List<Type> getInterfaceGenerics(Class interfaceType) {
    * @since 4.2
    */
   Optional<MethodElement> getMethod(String name, Class<?>... parameterTypes);
+
+  /**
+   * @return All the {@link MethodElement}s present in the class
+   * @since 4.5
+   */
+  List<MethodElement> getEnclosingMethods();
```

Comment:
per the usage it seems like these are always either iterated or transformed. Better to just return the `Stream<T>` to avoid the unnecesary `collect(toList())`